src/Commands/ChassisMotionProfileCommand.cpp

```diff
@@ -0,0 +1,249 @@
+#include <Commands/ChassisMotionProfileCommand.h>
+#include "Subsystems/Chassis.h"
+
+/**
+ * Notes about Talon Motion Profile
+ *
+ * The trajectory point buffer in the Talon Driver (RoboRio) is 2048 points.
+ * The typical trajectory point is 10ms. 2048 represents 20.48 seconds of motion profile.
+ *
+ * Important -- The code in this implemenation does not support more than 2048 trajectory points.
+ *
+ * The trajectory point buffer in the Talon (bottom buffer) holds 128 points.
+ *
+ * The Talon Motion Profile Mode has three drive modes controlled by Set()
+ * * Neutral (SetValueMotionProfileDisable)
+ * * Drive (SetValueMotionProfileEnable)
+ * * Hold (SetValueMotionProfileHold)
+ *
+ *
+ */
+
+namespace {
+
+  const double kSecondsToMillis = 1000.0;
+  const double kMillisToSeconds = ( 1.0 / kSecondsToMillis );
+
+  void LoadPoints(
+    const ChassisMotionProfileCommand::TrajectoryPoint* chassisRight,
+    const ChassisMotionProfileCommand::TrajectoryPoint* chassisLeft,
+    unsigned int trajectoryPointCount,
+    unsigned int pointDurationMillis)
+  {
+    unsigned int lastPoint = trajectoryPointCount - 1;
+
+    for (unsigned int point = 0; point < trajectoryPointCount; ++point) {
+      CANTalon::TrajectoryPoint rightTrajectoryPoint;
+      rightTrajectoryPoint.position = chassisRight[point].position;
+      rightTrajectoryPoint.velocity = chassisRight[point].velocity;
+      rightTrajectoryPoint.timeDurMs = pointDurationMillis;
+      rightTrajectoryPoint.profileSlotSelect = 1;   // always slot 1
+      rightTrajectoryPoint.velocityOnly = false;    // always both velocity and position
+      rightTrajectoryPoint.isLastPoint = (point == lastPoint);
+      rightTrajectoryPoint.zeroPos = (point == 0);
+
+      CANTalon::TrajectoryPoint leftTrajectoryPoint = rightTrajectoryPoint;
+      leftTrajectoryPoint.position = chassisLeft[point].position;
+      leftTrajectoryPoint.velocity = chassisLeft[point].velocity;
```

Comment:
`leftTrajectoryPoint` is constructed from `rightTrajectoryPoint`, so all of the members are set from the the values in `rightTrajectoryPoint`.

We don't do this very often with the robot. This is called a copy constructor. Construct a new variable with the values from the original variable. If no copy constructor is specified in the class, then the default copy constructor is used. The default copy constructor is a copy of each field of the original.